src/adservice/src/main/java/hipstershop/AdService.java

```diff
@@ -91,26 +92,20 @@ private void stop() {
     }
   }
 
-  static class AdServiceImpl extends hipstershop.AdServiceGrpc.AdServiceImplBase {
+  private static class AdServiceImpl extends hipstershop.AdServiceGrpc.AdServiceImplBase {
 
     /**
      * Retrieves ads based on context provided in the request {@code AdRequest}.
      *
      * @param req the request containing context.
-     * @param responseObserver the stream observer which gets notified with the value of
-     *     {@code AdResponse}
+     * @param responseObserver the stream observer which gets notified with the value of {@code
+     *     AdResponse}
      */
     @Override
     public void getAds(AdRequest req, StreamObserver<AdResponse> responseObserver) {
       AdService service = AdService.getInstance();
-      Span parentSpan = tracer.getCurrentSpan();
-      SpanBuilder spanBuilder =
-          tracer
-              .spanBuilderWithExplicitParent("Retrieve Ads", parentSpan)
-              .setRecordEvents(true)
-              .setSampler(Samplers.alwaysSample());
-      try (Scope scope = spanBuilder.startScopedSpan()) {
```

Comment:
This code was to demonstrate that you can further create child span within same process/thread. Here it is simply retrieving ads from a map but in real application it could be some expensive operation..

Reply:
ok.